### PR TITLE
Fix predefined parameter resolution

### PR DESCRIPTION
**Pull request by lordrhodos**
We need to set the type of the resolved predefined values otherwise a controller action like

```php
public function myAction(int fooId): Response {
...
}
```

willl throw a type error since we have enable strict types and parameters get injected as `string`

### CODE REVIEW

**Comment by felixsand** [comment]
Changing to "In progress" since tests are failing

File: src/Container/Container.php [hunk 1]
```diff
@@ -252,6 +252,9 @@ public function resolveParameters(array $parameters, array $predefinedValues = [
          */
         foreach ($parameters as $parameter) {
             if (\array_key_exists($parameter->getName(), $predefinedValues)) {
+                if ($parameter->getType()->isBuiltin()) {
```
Comment (felixsand) [inline src/Container/Container.php]:
```suggestion
                if ($parameter->getType() && $parameter->getType()->isBuiltin()) {
```
Since `$parameter->getType()` can return `null` if no type is specified. What do you think?

**Review by felixsand — CHANGES_REQUESTED** [review]
One question 🙂

**Comment by lordrhodos** [comment]
@PaulPylkas maybe you can check the tests and let me know if you want more tests for edge cases to be added that you've mentioned earlier?

**Review by felixsand — APPROVED** [review]
Excellent work! 🎉

**Review by PaulPylkas — APPROVED** [review]
:+1:
